zipline/algorithm.py

```diff
@@ -204,11 +204,11 @@ def __init__(self, *args, **kwargs):
         # Update the TradingEnvironment with the provided asset metadata
         self.trading_environment = kwargs.pop('env',
                                               TradingEnvironment.instance())
-        self.trading_environment.update_asset_finder(
-            asset_finder=kwargs.pop('asset_finder', None),
-            asset_metadata=kwargs.pop('asset_metadata', None),
-            identifiers=kwargs.pop('identifiers', None)
+        self.trading_environment.write_data(
```

Comment:
If the desired pattern now is to write the metadata directly to the environment, should we just go ahead and scrap the ability to write metadata here?

I still do see a use case for providing identifiers -- its the simplest way to spin up an algorithm. However, if the user wants to provide more metadata about their assets, we might now be wanting them to do it directly.

Also, here we are assuming that all metadata coming in is equity data?